### PR TITLE
various fixes to generator execution

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/shared/exec-util.c [hunk 1]
```diff
@@ -36,27 +36,35 @@
 /* Put this test here for a lack of better place */
 assert_cc(EAGAIN == EWOULDBLOCK);
 
-static int do_spawn(const char *path, char *argv[], int stdout_fd, pid_t *pid, bool set_systemd_exec_pid) {
-        pid_t _pid;
+static int do_spawn(
+                const char *path,
+                char *argv[],
+                int stdout_fd,
+                pid_t *ret_pid,
```
Comment (YHNdnzj) [inline src/shared/exec-util.c]: If you touch this, move the ret_ param to the end?

**Comment by poettering** [comment]
The last commit is I guess a work-around for https://github.com/libbpf/libbpf/issues/812, but really should be fixed in libbpf, it's not OK to leak non-O_CLOEXEC fds into PID 1 really.

**Review by keszybz — APPROVED** [review]

**Comment by bluca** [comment]
Givent the more targeted https://github.com/systemd/systemd/pull/33074 this sounds like material for the next release?

**Comment by poettering** [comment]
Yeah, i intended this for v257

**Comment by keszybz** [comment]
I tested this in Fedora rawhide, and it indeed fixes the selinux avc.
I force pushed with the parameter order switch that was requested above.

**Comment by bluca** [comment]
You mean you want this in 256 too? If so, could you open a separate PR with just the last commit?

**Comment by keszybz** [comment]
I think it should be fine to merge all three. They were tested as such and the third commit touches the same code as the second one.

**Comment by keszybz** [comment]
All rawhide builds are failing because the switch to dnf5 just happened and the system is trying to install two different versions of dnf5 for some reason. I expect it'll be fixed quickly.

**Comment by keszybz** [comment]
I want to merge this after the CI passes. The rawhide builds are in the process of being fixed, so I'll restart those builds shortly. If anyone is against, please speak up. The motivation for merging is that libbpf and selinux are fairly widespread and this will cause noticable failures for users. The patch is doing the right thing anyway, making our code more resilient. If anyone is against merging, please speak up.

**Review by bluca — APPROVED** [review]

**Review by YHNdnzj — APPROVED** [review]

**Review by yuwata — COMMENTED** [review]
LGTM.

**Pull request merged**